core/src/test/kotlin/com/kin/ecosystem/core/accountmanager/AccountManagerImplTest.kt

```diff
@@ -0,0 +1,152 @@
+    private val blockchainEvents: BlockchainEvents = mock {
+        on { addAccountCreationListener(any()) } doAnswer { accountCreationRegistration }
+    }
+    private var kinAccount: KinAccount = mock {
+        on { blockchainEvents() } doAnswer { blockchainEvents }
+    }
+
+
+    private lateinit var accountManager: AccountManager
+
+    @Before
+    override fun setUp() {
+        super.setUp()
+        resetInstance()
+    }
+
+    @Throws(Exception::class)
+    private fun resetInstance() {
+        val instance = AccountManagerImpl::class.java.getDeclaredField("instance")
+        instance.apply {
+            isAccessible = true
+            set(null, null)
+        }
+    }
+
+    private fun initWithFirstState(@AccountState state: Int) {
+        whenever(local.accountState).thenReturn(state)
+        AccountManagerImpl.init(local, eventLogger, authRepository, blockchainSource)
+        accountManager = AccountManagerImpl.getInstance()
+    }
+
+    @Test
+    fun `getAccountState should return the current state`() {
+        initWithFirstState(REQUIRE_CREATION)
+        assertEquals(REQUIRE_CREATION, accountManager.accountState)
+
+        whenever(local.accountState).doReturn(PENDING_CREATION)
+        assertEquals(PENDING_CREATION, accountManager.accountState)
+
+        whenever(local.accountState).doReturn(REQUIRE_TRUSTLINE)
+        assertEquals(REQUIRE_TRUSTLINE, accountManager.accountState)
+
+        whenever(local.accountState).doReturn(CREATION_COMPLETED)
+        assertEquals(CREATION_COMPLETED, accountManager.accountState)
+    }
+
+    @Test
+    fun `isAccountCreate should return false`() {
+        initWithFirstState(REQUIRE_CREATION)
+        whenever(local.accountState).doReturn(REQUIRE_CREATION)
+        assertFalse(accountManager.isAccountCreated)
+
+        whenever(local.accountState).doReturn(PENDING_CREATION)
+        assertFalse(accountManager.isAccountCreated)
+
+        whenever(local.accountState).doReturn(REQUIRE_TRUSTLINE)
+        assertFalse(accountManager.isAccountCreated)
+    }
+
+    @Test
+    fun `isAccountCreate should return true`() {
+        initWithFirstState(REQUIRE_CREATION)
+        whenever(local.accountState).doReturn(CREATION_COMPLETED)
+        assertTrue(accountManager.isAccountCreated)
+    }
+
+    @Test
+    fun `start account creation flow and listen for state changes`() {
+        initWithFirstState(REQUIRE_CREATION)
+        val stateList = arrayListOf<Int>()
+        val stateObserver = object : Observer<Int>() {
+            override fun onChanged(value: Int) {
+                stateList.add(value)
+            }
+        }
+
+        accountManager.addAccountStateObserver(stateObserver)
+        accountManager.start()
+        argumentCaptor<EventListener<Void>>().apply {
+            verify(blockchainEvents).addAccountCreationListener(capture())
+            firstValue.onEvent(null)
+        }
+
+
+
+        assertEquals(listOf(1, 1, 2, 3, 4), stateList)
+    }
+
+
```

Comment:
What, you're not following a set of conventions? Why two empty lines here, but above you only had one?